pkg/types/encrypted_private_key.go

```diff
@@ -112,9 +113,40 @@ func (e *EncryptedPrivateKey) PublicKey(password *memguard.LockedBuffer, salt, n
 	}, nil
 }
 
-func PrivateKey(password *memguard.LockedBuffer, salt, nonce, encryptedKey []byte) (*memguard.LockedBuffer, error) {
-	aeadCipher, secretKey, err := crypto.NewSecretCipher(password.Bytes(), salt[:])
-	defer password.Destroy()
+// PrivateKeyTextInClear returns the private key in clear. Guarded password is destroyed.
+func (e *EncryptedPrivateKey) PrivateKeyTextInClear(guardedPassword *memguard.LockedBuffer, salt, nonce, encryptedKey []byte) (*memguard.LockedBuffer, error) {
+	privateKeyInClear, err := PrivateKey(guardedPassword, salt, nonce, encryptedKey)
+	if err != nil {
+		return nil, err
+	}
+
+	seed := ed25519.PrivateKey(privateKeyInClear.Bytes()).Seed()
```

Comment:
privateKeyInClear.Bytes() has the same value as seed.Bytes(), why do you need to duplicate it?

Reply:
It's weird, looking at the code of the Seed function, it should return the private key bytes: https://cs.opensource.google/go/go/+/refs/tags/go1.21.3:src/crypto/ed25519/ed25519.go;l=75